src/db/error.rs

```diff
@@ -8,4 +8,6 @@ use thiserror::Error;
 pub enum MongoDbError {
     #[error("database error: {0}")]
     DatabaseError(#[from] mongodb::error::Error),
+    #[error("failed to serialize to BSON: {0}")]
+    BsonSerializationError(#[from] mongodb::bson::ser::Error),
 }
```

Comment:
Should we use `#[error(transparent)` here too?

Reply:
I think in this case the context may be important.